rir/src/compiler/translations/pir_2_rir/pir_2_rir.cpp

```diff
@@ -919,6 +919,19 @@ rir::Code* Pir2Rir::compileCode(Context& ctx, Code* code) {
                       needsEnsureNamed.count(instr)))
                 cb.add(BC::ensureNamed());
 
+            // Check the return type
```

Comment:
why do you only check the return type?

can you have different check levels, and on level 2 you also check let's say binops, or call builtin arguments/results

Reply:
I think arg types are already checked, because the verifier checks after each pass that each arg's return type conforms to its instructions' expected arg type, and when the arg instruction is executed it must conform to its return type

Reply:
> each arg's return type conforms to its instructions' expected arg type, 

but it could still be wrong. that's just the static type of the arguments. this pr is all about verifying that the static type actually corresponds to the dynamic type.

so e.g. if you have `a+b` and we think `a` is an int, but it isn't actually, we might not notice. so it would be useful to check it.

> I'm not sure what you mean by builtin results

if you have a callBuiltin and we somehow think it returns a certain type, it would be nice to check that it actually does. same with binops and so on